cpp/unreal_projects/SpearSim/Source/SpearSim.Target.cs

```diff
@@ -14,7 +14,7 @@ public SpearSimTarget( TargetInfo Target) : base(Target)
         DefaultBuildSettings = BuildSettingsVersion.V2;
         ExtraModuleNames.AddRange(new string[] { "SpearSim" });
 
-        if (Target.Platform == UnrealTargetPlatform.Win64) {
+        if (Target.Platform == UnrealTargetPlatform.Win64 || Target.Platform == UnrealTargetPlatform.Win32) {
```

Comment:
Why do we need this check for `SpearSim.Target.cs` but not for any of our other C# files?

Reply:
I think `GenerateProjectfiles` feature requires this addition only at the project level.

Reply:
Without this addition, this is the error
```bash
ERROR: Unable to instantiate instance of 'SpearSimTarget' object type from compiled assembly 'SpearSimModuleRules'.  Unreal Build Tool creates an instance of your module's 'Rules' object in order to find out about your module's requirements.  The CLR exception details may provide more information:  System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Exception: [SPEAR | SpearSim.Target.cs] Target.Platform == Win32
          at SpearSimTarget..ctor(TargetInfo Target) in c:\github\spear\cpp\unreal_projects\SpearSim\Source\SpearSim.Target.cs:line 47
          --- End of inner exception stack trace ---
          at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
          at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
          at UnrealBuildTool.RulesAssembly.CreateTargetRulesInstance(String TypeName, TargetInfo TargetInfo) in D:\Build\++UE4\Sync\Engine\Source\Programs\UnrealBuildTool\System\RulesAssembly.cs:line 538
```

Reply:
I see. In that case, we should include a comment.

```
// We need to include a check for UnrealTargetPlatform.Win32 here (but not in SpearSimEditorTarget), otherwise we get an error when running Unreal's GenerateProjectFiles.bat script.
```